### PR TITLE
feat: add network protocol config

### PR DESCRIPTION
**Pull request by quake**
This feature is designed for testing and use of ckb-network crate by third party code, allows user to customize network protocols, for example, only open sync protocol in full node:

```
support_protocols = ["Sync"]
```

### CODE REVIEW

**Comment by nervos-bot-user** [comment]
Can one of the admins verify this patch?

File: network/src/network.rs [hunk 1]
```diff
@@ -738,65 +738,83 @@ impl<T: ExitHandler> NetworkService<T> {
...
             ProtocolHandle::Callback(Box::new(IdentifyProtocol::new(identify_callback)))
         });
+        protocol_metas.push(identify_meta);
 
-        // Feeler protocol
-        let feeler_meta = SupportProtocols::Feeler.build_meta_with_service_handle({
-            let network_state = Arc::clone(&network_state);
-            move || ProtocolHandle::Both(Box::new(Feeler::new(Arc::clone(&network_state))))
-        });
+        // TODO: how to deny banned node to open those protocols?
```
Comment (driftluo) [inline network/src/network.rs:757]: The premise of opening any protocol is to establish a connection. This step will check the ban list, if i remember correctly

Comment (quake) [reply network/src/network.rs]: removed misleading comment line, please review again, thanks.

**Review by driftluo — DISMISSED** [review]

**Review by zhangsoledad — APPROVED** [review]

**Comment by quake** [comment]
rebased with develop, please review again.

**Comment by zhangsoledad** [comment]
https://github.com/nervosnetwork/ckb/pull/2857

**Review by driftluo — APPROVED** [review]

**Comment by doitian** [comment]
bors r=zhangsoledad,driftluo

**Pull request merged**